File: app/src/main/java/com/example/Sleepy/classes/AppTheme.java
package com.example.Sleepy.classes;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.appcompat.app.AppCompatDelegate;

import com.example.Sleepy.R;

public class AppTheme {
    public static void setShareTheme(Context context) {
        SharedPreferences prefs = context.getSharedPreferences("SETTINGS", Context.MODE_PRIVATE);
        switch (prefs.getInt("THEME", R.id.rbThemeAuto)) {
            case R.id.rbThemePurple:
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                break;
            case R.id.rbThemeDark:
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                break;
            case R.id.rbThemeAuto:
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
                break;
        }
    }
}

File: app/src/main/java/com/example/sleepy/utils/AppTheme.kt
package com.example.sleepy.utils

import android.content.Context
import com.example.sleepy.R
import androidx.appcompat.app.AppCompatDelegate
import com.example.sleepy.data.storage.PrefsStorage

class AppTheme {
    companion object{
        fun setShareTheme(context: Context) {
            val preferences = PrefsStorage(context)
            when (preferences.themeId) {
                R.id.rbThemePurple -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                R.id.rbThemeDark -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                R.id.rbThemeAuto -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
            }
        }
    }

}